code/script.js

```diff
@@ -15,37 +20,190 @@ const showMessage = (message, sender) => {
+  event.preventDefault();
+  userName = nameInput.value;
+  nameInput.value = "";
+  showMessage(`I'm ${userName}.`, 'user');
+  setTimeout(drinkTypeOptions, 1500);
+}
+
+document.getElementById("nameForm").addEventListener("submit", userNameInput);
+
+// Bot responds by welcoming the user to Hot Java Coffee and asks what the user would like to drink.
+const drinkTypeOptions = () => {
+  showMessage(`Well hello ${userName}! The warmest of welcomes to Hot Java Coffee.
+  What are you in the mood for today?`, 'bot');
+  // Buttons for the different drink choices
+    inputWrapper.innerHTML = `
+    <button id="caffeLatte">Caffe Latte</button>
+    <button id="flatWhite">Flat White</button>
+    <button id="cappuccino">Cappuccino</button>`
+
+      document.getElementById("caffeLatte").addEventListener("click", () => {
+        drinkType = "Caffe Latte"; 
+        showMessage("I'd love a Caffe Latte!", 'user');
+        setTimeout(milkOptions, 1500);
+      });
+      document.getElementById("flatWhite").addEventListener("click", () => {
+        drinkType = "Flat White"; 
+        showMessage("Flat White is my jam!", 'user');
+        setTimeout(milkOptions, 1500);
+      });
+      document.getElementById("cappuccino").addEventListener("click", () => {
+        drinkType = "Cappuccino"; 
+        showMessage("I've never said 'no' to a Cappuccino!", 'user');
+        setTimeout(milkOptions, 1500);
+      });
+}
+
+  // When the drink is chosen, the bot asks the user which milk they want for their drink.
+const milkOptions = () => {
+  showMessage(
+    `A ${drinkType}? Lovely! What kind of milk would you like us to prepare it with?`,
+    'bot');
+    // Buttons for the different milk choices
+      inputWrapper.innerHTML = `
+      <button id="oatMilk">Oat milk</button>
+      <button id="almondMilk">Almond milk</button>
+      <button id="soyMilk">Soy milk</button>`
+
+        document.getElementById("oatMilk").addEventListener("click", () => {
+          milkType = "Oat milk"; 
+          showMessage("Oat milk, please!", 'user');
+          setTimeout(confirmMilk, 1500);
+        });
+        document.getElementById("almondMilk").addEventListener("click", () => {
+          milkType = "Almond milk"; 
+          showMessage("Almond milk, please!", 'user');
+          setTimeout(confirmMilk, 1500);
+        });
+        document.getElementById("soyMilk").addEventListener("click", () => {
+          milkType = "Soy milk"; 
+          showMessage("Soy milk, please!", 'user');
+          setTimeout(confirmMilk, 1500);
+        });
+}
+
+// Bot confirms the choice of milk to the user
+const confirmMilk = () => {
+  showMessage(`You chose to have your drink with ${milkType}. Great choice!`, 'bot');
+  setTimeout(sizeChoice, 2000);
+}
+
+
+// Bot asks the user which size of the drink they want
+const sizeChoice = () => {
+  showMessage(
+    `Which drink size would you prefer today: big or small?`,
+    'bot')
+    // Buttons for the drink sizes
```

Comment:
Reading your code gets very easy whith all the comments. They're very accurate and "to the point".